### ISSUE TITLE
Create a custom plug-in

### ISSUE THREAD

**Issue by GBKS** [GBKS/bitcoin-wallet-ui-kit#49]
I already built some custom plug-ins for maintaining the UI Kit, like the [screen export helper](https://github.com/GBKS/bitcoin-wallet-ui-kit/tree/main/figma-plugins/screen-export-helper), which automatically preps screens in the Figma file export and rendering on the [website](https://www.bitcoinuikit.com/screens). We also us the [Lights](https://www.figma.com/community/plugin/780821534053786200/lights) plug-in for switching themes. I think it might be worth creating a custom plug-in for the UI Kit and publishing it to Figma community.

Design feature ideas:
- Theme switcher (select 1+ screens, click to toggle)
- Auto-format bitcoin/sats (select a text field, pick currency & locale, click to format)
- Auto bitcoin/fiat conversion
- QR code generator
- Search the design guide for info (not sure how to technically do this)

Maintenance feature ideas:
- Find instances whose component is external or deleted (done)
- Find instances of a main component (done)
- Find text without a text style

Admin tools:
- Prep screens for export to website

I already built a basic plug-in that does some of this, but it's more of a prototype. I'd like to wrap it up and get it out there.

Visually, I imagine something like this. I thought of making this way more playful, but I think plug-ins should not overshadow the actual design you're working on.

![image](https://github.com/GBKS/bitcoin-wallet-ui-kit/assets/695901/86b9f3ec-75c6-4c06-a0dc-d19e6b651176)

What other useful tools could we provide? Links to the website? [ChatBTC](https://chat.bitcoinsearch.xyz) integration?

**Comment by GBKS** [comment]
Just realized I already mentioned this [here](https://github.com/GBKS/bitcoin-wallet-ui-kit/issues/23#issuecomment-1606038959) in June.

**Comment by GBKS** [comment]
[Prototype demo video](https://youtu.be/GOKCVeI2gn0) of this plug-in, showing a new idea for theme presets that can be instantly applied through the whole kit with a few clicks.

**Comment by GBKS** [comment]
Thinking through this a little more, it looks like we don't need a separate light and dark mode anymore if we have these style presets. Light/dark are simply two presets. That simplifies the basic color system a bit further. There's just one set of colors, and you assign, for example, the "Default light" theme or the "Default dark" theme, etc.

The color system might then be:
- Primary
- Accent (set of 12 colors - the rainbow)
- Neutrals (10 colors - black to white)

That means the website will also need to be updated accordingly. If we can come up with a set of 3-4 color themes, typographic sets, etc, we can also add them on the [foundation page](https://www.bitcoinuikit.com/foundation).

I also think that plug-in warrants its own page on the website then. It can include a how-to video and we can link to it from the plug-in itself in case people get stuck.

**Comment by GBKS** [comment]
The color palette changes are now listed in a separate issue #51.

**Comment by GBKS** [comment]
I published this to Figma community. They'll review it, which can take 5-10 days. You can [get the code here](https://github.com/GBKS/bitcoin-wallet-ui-kit/tree/main/figma-plugins/bitcoin-ui-kit). I also published an updated Figma file with all the clean-up, simplified and expanded color palette and new themes.

I'll close this issue once the Figma review is complete.

**Comment by GBKS** [comment]
That was fast, the plugin was already reviewed and is now [available in Figma community](https://www.figma.com/community/plugin/1254869452790045259/bitcoin-ui-kit). Woohoo! I'll need to put together a short video about it and create some website content. That falls under #3 and #52, so going to close this issue.
